### ISSUE TITLE
Metadata issue under Windows

### ISSUE THREAD

**Issue by DavidFricker** [SimonBlanke/Hyperactive#5]
Hi, thank you for making this amazing package!

When I try to run the demo (with no changes) on a windows environment under conda using the latest version from pypi I run into the following error. Please could you tell me what is going wrong or if windows is even a supported OS? 

Error:
```
(hyperactive) λ python test.py
Traceback (most recent call last):
  File "test.py", line 24, in <module>
    opt.search(search_config, n_iter=20)
  File "C:\Users\<user>\Miniconda3\lib\site-packages\hyperactive\hyperactive_api.py", line 40, in search
    core.run()
  File "C:\Users\<user>\Miniconda3\lib\site-packages\hyperactive\hyperactive_core.py", line 20, in run
    dist.dist(Search, self._main_args_, self._opt_args_)
  File "C:\Users\<user>\Miniconda3\lib\site-packages\hyperactive\distribution.py", line 31, in dist
    self.dist_default(search_class, _main_args_, _opt_args_)
  File "C:\Users\<user>\Miniconda3\lib\site-packages\hyperactive\distribution.py", line 36, in dist_default
    _optimizer_.search()
  File "C:\Users\<user>\Miniconda3\lib\site-packages\hyperactive\search.py", line 73, in search
    self._run_job(nth_process)
  File "C:\Users\<user>\Miniconda3\lib\site-packages\hyperactive\search.py", line 96, in _run_job
    _cand_, _p_ = self._search(nth_process)
  File "C:\Users\<user>\Miniconda3\lib\site-packages\hyperactive\search.py", line 122, in _search
    _cand_ = self._initialize_search(self._main_args_, nth_process, self._info_)
  File "C:\Users\<user>\Miniconda3\lib\site-packages\hyperactive\search.py", line 146, in _initialize_search
    _cand_ = Candidate(nth_process, _main_args_, _info_)
  File "C:\Users\<user>\Miniconda3\lib\site-packages\hyperactive\candidate.py", line 49, in __init__
    self.mem = LongTermMemory(self._space_, _main_args_, self)
  File "C:\Users\<user>\Miniconda3\lib\site-packages\hyperactive\memory\memory.py", line 39, in __init__
    self._load_ = MemoryLoad(_space_, _main_args_, _cand_)
  File "C:\Users\<user>\Miniconda3\lib\site-packages\hyperactive\memory\memory_load.py", line 25, in __init__
    super().__init__(_space_, _main_args_, _cand_)
  File "C:\Users\<user>\Miniconda3\lib\site-packages\hyperactive\memory\memory_io.py", line 35, in __init__
    os.makedirs(self.date_path, exist_ok=True)
  File "C:\Users\<user>\Miniconda3\lib\os.py", line 211, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "C:\Users\<user>\Miniconda3\lib\os.py", line 211, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "C:\Users\<user>\Miniconda3\lib\os.py", line 211, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "C:\Users\<user>\Miniconda3\lib\os.py", line 221, in makedirs
    mkdir(name, mode)
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Users\\<user>\\Miniconda3\\lib\\site-packages\\hyperactive\\memory\\paths.py/meta_data'
```

Code being run in `test.py`
```
from sklearn.model_selection import cross_val_score
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.datasets import load_breast_cancer
from hyperactive import Hyperactive

data = load_breast_cancer()
X, y = data.data, data.target

'''define the model in a function'''
def model(para, X, y):
    '''optimize one or multiple hyperparameters'''
    gbc = GradientBoostingClassifier(n_estimators=para['n_estimators'])
    scores = cross_val_score(gbc, X, y)

    return scores.mean()

'''create the search space and search_config'''
search_config = {
    model: {'n_estimators': range(10, 200, 10)}
}

'''start the optimization run'''
opt = Hyperactive(X, y)
opt.search(search_config, n_iter=20)
```

Versions:
 - Python 3.7.3
 - hyperactive 2.3.0
 - sklearn 0.21.3
 - Windows 10 64bit

**Comment by SimonBlanke** [comment]
Hello @DavidFricker 

thank you for reporting this issue. 

There is currently no testing in Windows. Therefore Windows is officially not a supported OS. But there is a simple workaround.
Just change memory to "short":

`
opt = Hyperactive(X, y, memory="short")
`

Explanation:
The bug occurs because Hyperactive wants to save the "long term memory" to disk. This is a feature that automatically saves optimization data for later use.
Windows uses backslashes for the path, while Ubuntu uses forward slashes.
I will try to find a fix for this problem for Hyperactive v3.0.0

Thanks again for the feedback!

**Comment by DavidFricker** [comment]
Thank you for the fast clarification and solution! It seems to be working with the `short` memory param

Perhaps using `os.path.join` would be good solution to this problem? As I understand it, this function will use the correct slash for the OS the code is being run on.

Thanks again 😃
